### PR TITLE
[Merged by Bors] - add metrics for conservative cache and hare

### LINKED ISSUE #3365: conservative cache: add metrics
to measure time spent in
- add tx to cache
- gathering mempool txs
- applying a block
....

### PR DESCRIPTION
**Pull request by countvonzero**
## Motivation

Closes #3365

## Changes

- best effort parsing synced block txs
- add metrics for conservative cache and hare

## Test Plan

unit tests, systests

### CODE REVIEW

**Comment by countvonzero** [comment]
@dshulyak please review. in the first commit i added parsing for the synced block txs. thanks

File: hare/hare.go [hunk 1]
```diff
@@ -229,37 +229,47 @@ func (h *Hare) collectOutput(ctx context.Context, output TerminationOutput) erro
...
 	} else {
+		consensusCount.WithLabelValues(failure).Inc()
 		h.WithContext(ctx).With().Info("hare terminated with failure", layerID)
 	}
 
 	hareOutput := types.EmptyBlockID
 	if len(pids) > 0 {
 		// fetch proposals from peers if not locally available
 		if err := h.fetcher.GetProposals(ctx, pids); err != nil {
+			blockGenCount.WithLabelValues(failFetch).Inc()
```
Comment (dshulyak) [inline hare/hare.go]: i think it is better to initialize this metrics with a label and then use it here, e.g.

```
var failFetchCount = blockGenCount.WithLabelValues...
failFetchCount.Inc()
```

**Review by dshulyak — APPROVED** [review]

File: hare/hare.go (hunk 1 above)
Comment (countvonzero) [reply hare/hare.go]: thanks. done for the metrics in this pkg

**Comment by countvonzero** [comment]
bors merge

**Pull request closed without merging**